### ISSUE TITLE
Existing wallets with no `walletConfig` always request Gaia on unlock

### ISSUE THREAD

**Issue by kyranjamie** [blockstack/blockstack-app#2137]
I noticed that on an existing account with no `walletConfig` set, Gaia is always called slowing down the unlock. This happens as localStorage is only set when creating a new wallet, or a new account. I hadn't done either, so it was slow to unlock.

Rather than addressing this issue directly, I think we should reevaluate using localstorage here at all. It's bug-prone, adds a handful of difficult-to-test edge cases, and isn't really necessary.

The objective of https://github.com/hirosystems/stacks-wallet-web/pull/2080 was to prevent the Gaia call from slowing down the unlock. This could alternatively be achieved by calling the same method immediately after unlock, so that it doesn't contribute to the unlock time, instead working silently in the background. This would be a much simpler implementation, and wouldn't involve:
- Using localStorage (handling error/fallback cases)
- Importing the `bip32` and `bitcoinjs-lib` (breaking wallet-generation encapsulation of `wallet-sdk`)
- All the new functions added

Much of this only be came apparent to me after the fact, so didn't comment this on PR, nonetheless we should definitely come up with a much simpler solution to this problem.

cc/ @beguene

**Comment by beguene** [comment]
With my last fix https://github.com/hirosystems/stacks-wallet-web/pull/2128, when an existing account has no walletConfig, on the **first** unlock the gaia wallet config is retrieved and stored, then the subsequent unlocks won't have any gaia fetch so the unlock time should be relatively small. My fix was already merged, so this issue should not occur on dev branch. 
Let me know if you still face this issue and the steps to reproduce in that case.

**Comment by kyranjamie** [comment]
> on the first unlock the gaia wallet config is retrieved and stored

Not as far as I can tell. Check the unlock method. There is no code here, or in the methods it calls, to set localstorage.

https://github.com/hirosystems/stacks-wallet-web/blob/1882c1f101f1a0be2d52e2df0d8cc1dead9fbb0c/src/background/vault.ts#L151-L179

To replicate:
1) Create a wallet
2) Delete `walletConfig` from localStorage (to replicate older wallet version)
3) Reload extension from the extensions panel to ensure both scripts are refreshed
4) Open background panel to see network requests
5) Unlock wallet --> gaia request made, no localStorage set
6) Lock and unlock wallet --> same behaviour, gaia request made, no localStorage set

As I explain, though, I don't think the solution here is to patch this edge case. Rather, come up with a simpler alternate solution that doesn't cache the whole config in localstorage. What are your thoughts on doing this? 

I have some ideas for how we could cache just the highest account index, and come up with the same result as now, without having to reimplement functionality of wallet-sdk, or use localstorage to cache a json object. I'd be happy to talk through them.

**Comment by beguene** [comment]
Indeed it's an issue, I was able to reproduce. It does not store it in localStorage in that case, I got confused with the storeSeed function.
I think your solution with the highest index works well, but the reason I stored the full gaia json is to be able to remove some other gaia dependencies and calls in other parts of the code later, for example in displaying .btc names etc.. 
The json acts like an interface so the other parts of the code won't need much changes.
